backend/database/migrations/000001-create-user-table.up.sql

```diff
@@ -3,7 +3,7 @@ BEGIN;
 CREATE TABLE IF NOT EXISTS user_ (
 	id         SERIAL PRIMARY KEY,
 	username   VARCHAR (50) UNIQUE NOT NULL,
-	password   VARCHAR (50) NOT NULL  
+	password   VARCHAR (70) NOT NULL  
```

Comment:
"just for future reference, it's better to create a new migration rather than updating an existing migration. it'll break evns that have already ran this, especially if someone like me didn't catch the update and reran this migration" Not 100% sure what you mean by that as you should just be able to run the migrate up file and it will create a blank table for you? Since we dont have any data migrated for it. 

I have been sending it through post man.

Reply:
so in the sense of running migration files, you run each once per project typically, they never get re-ran.

let's say in the context that we're 20 migrations in, and we decided to update the user table's VARCHAR len for the password just like we're doing here.

if we update the original migration file like we're doing here, most migration management systems will not capture the update. this can lead to weird bugs that we end up not understanding.

the correct way to do this is by creating a new migration file (it would be unregistered to the migration management system, which means that it knows that a new migration is on the docket, and needs to be ran).

it would be something similar to 000021-change-password-length.up.sql

```sql
BEGIN;

ALTER TABLE user_ ALTER COLUMN password TYPE VARCHAR(70);

COMMIT;
```

Also, an aside:
gotta be mindful of the migrations, for example, this works well because we're _increasing_ the VARCHAR length, rather than decreasing.

if we were decreasing and had some user_'s with passwords that were at the 70 length and tried reducing the size, it would break the table / db